src/DocumentFormat.OpenXml/Packaging/SpreadsheetDocument.cs

```diff
@@ -119,7 +120,15 @@ private void UpdateDocumentTypeFromContentType()
         /// <exception cref="ArgumentNullException">Thrown when "path" is null reference.</exception>
         public static SpreadsheetDocument Create(string path, SpreadsheetDocumentType type)
         {
-            return Create(path, type, true);
+            SpreadsheetDocument smlDoc = Create(path, type, true);
```

Comment:
I would rather it not be injected here. I would like to see some sort of service that would be run at creation time. Something like

```csharp
public interface IPackageInitializer
{
  void Initialize(OpenXmlPackage package);
}
```

You could inject it via OpenSettings

Reply:
Ok, we'll take some more time to think about this. Upon reflection, this should not be done in SpreadsheetDocument anyway. I've created a project for this so we don't lose track.  https://github.com/OfficeDev/Open-XML-SDK/projects/9

Reply:
Whatever magic we add here should not create breaking changes. When creating documents with the SDC, no parts are added at the moment. However, it would certainly be nice to be able to tell the SDK to create more "complete" documents such as minimal Word, Excel, and PowerPoint documents that fulfill certain criteria. But those should have to be created explicitly (e.g., through settings or otherwise).